deploy/post_deploy/nsx_cluster.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# We get this passed from the main script
+NSX_MANAGER=$1
+NSX_CONTROLLER=$2
+NSX_SERVICE=$3
+KVM_HOST=$4
+
+NSX_CONTROLLER_IP=$(getent hosts ${NSX_CONTROLLER} | awk '{ print $1 }')
+NSX_SERVICE_IP=$(getent hosts ${NSX_SERVICE} | awk '{ print $1 }')
+KVM_HOST_IP=$(getent hosts ${KVM_HOST} | awk '{ print $1 }')
+
+# Wait for controller to be responding
+while ! ping -c3 ${NSX_CONTROLLER} &>/dev/null; do
+  sleep 2
+done
+
+SSH_OPTIONS="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -q"
+
+# Create cluster
+echo "Note: Creating NSX cluster"
+sudo yum install -y -q sshpass
+sshpass -p 'admin' ssh ${SSH_OPTIONS} admin@${NSX_CONTROLLER_IP} join control-cluster ${NSX_CONTROLLER_IP}
+sshpass -p 'admin' ssh ${SSH_OPTIONS} admin@${NSX_SERVICE_IP} set switch manager-cluster ${NSX_CONTROLLER_IP}
+
+# login to controller
+echo "Note: Authenticating against controller"
+curl -k -c cookie.txt -X POST -d 'username=admin&password=admin' https://${NSX_CONTROLLER}/ws.v1/login
```

Comment:
Without any kind of context whatsoever, but just noticed that there is a hard coded password here while all else is parameterized.